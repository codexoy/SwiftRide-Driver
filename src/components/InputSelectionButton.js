import React from "react"
import { StyleSheet, View, TouchableHighlight } from "react-native"
import { colors } from "../consts"

export default class InputSelectionButton extends React.Component {
	render() {
		let style = { ...styles.rects, ...this.props.style }
		if (this.props.selected || this.props.secondTimeSelected) {
			const selectedColor = this.props.selectedColor || colors.blue
			style = {
				...style,
				...styles.selected,
				backgroundColor: selectedColor,
				shadowColor: selectedColor
			}
		}
		if (this.props.secondTimeSelected) {
			style = { ...style, ...styles.secondTimeSelected }
		}
		return (
			<TouchableHighlight
				onPress={this.props.onPress}
				underlayColor="#ffffff00"
				activeOpacity={0.3}
			>
				<View style={style}>{this.props.children}</View>
			</TouchableHighlight>
		)
	}
}

const styles = StyleSheet.create({
	rects: {
		margin: 6,
		padding: 6,
		backgroundColor: "#f8f8f8",
		borderRadius: 4,
		maxWidth: 140,
		alignItems: "center"
	},
	selected: {
		borderRadius: 4,
		shadowOffset: {
			width: 0,
			height: 6
		},
		shadowOpacity: 0.2,
		shadowRadius: 6,
		elevation: 4
	},
	secondTimeSelected: {
		backgroundColor: "green"
	}
})
