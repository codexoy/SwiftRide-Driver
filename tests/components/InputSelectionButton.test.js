import { Text } from "react-native"
import React from "react"

import renderer from "react-test-renderer"

import InputSelectionButton from "../../src/components/InputSelectionButton"

describe("InputSelectionButton", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<InputSelectionButton onPress={() => "test"} selected={true}>
					<Text>test</Text>
				</InputSelectionButton>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
