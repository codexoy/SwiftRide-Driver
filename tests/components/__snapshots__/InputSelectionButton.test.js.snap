// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSelectionButton view renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(12,116,244)",
        "borderRadius": 4,
        "elevation": 4,
        "margin": 6,
        "padding": 6,
        "shadowColor": "rgb(12,116,244)",
        "shadowOffset": Object {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 6,
      }
    }
  >
    <Text>
      test
    </Text>
  </View>
</View>
`;
